test/hotspot/jtreg/compiler/c2/gvn/TestCountBitsRange.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2025 Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2.gvn;
+
+import compiler.lib.ir_framework.*;
+
+/*
+ * @test
+ * @bug 8360192
+ * @summary Tests that count bits nodes are handled correctly.
+ * @library /test/lib /
+ * @requires vm.compiler2.enabled
+ * @run driver compiler.c2.gvn.TestCountBitsRange
+ */
+public class TestCountBitsRange {
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    static int i = RunInfo.getRandom().nextInt();
+    static long l = RunInfo.getRandom().nextLong();
+
```

Comment:
It would be nice to also check the return values of the functions with a non-compiled version, so that we can make sure that the constant folding results are correct as well.